src/audio_core/cubeb_sink.cpp

```
@@ -27,6 +31,11 @@ struct CubebSink::Impl {
 };
 
 CubebSink::CubebSink(std::string_view target_device_name) : impl(std::make_unique<Impl>()) {
+    // Cubeb requires COM to be initialized on the thread calling cubeb_init on Windows
+#ifdef _MSC_VER
+    com_init_result = CoInitializeEx(nullptr, COINIT_MULTITHREADED);
```

This might not interact well with the existing single-threaded `CoInitialize`:
https://github.com/citra-emu/citra/blob/acaca4188e62820242f1ee902400bde4d8350dcd/src/audio_core/hle/wmf_decoder.cpp#L35

cc @liushuyu

---

For some reason, if I use 
```cpp
CoInitializeEx(NULL, COINIT_MULTITHREADED);
```
In `wmf_decoder.cpp`, the result will be `RPC_E_CHANGED_MODE` indicating another object got initialized using another threading model. After fiddling around a bit, I found the compatible flag was `COINIT_APARTMENTTHREADED`. Not sure which object initialized prior to `wmf_decoder` with such flag (probably Cubeb but I can't test that during debugging)

---

@liushuyu The current cubeb version we uses has a call to `HRESULT hr = CoInitializeEx(nullptr, COINIT_MULTITHREADED);` in cubeb_wasapi.cpp

---

Qt likely is the one that makes incompatible call: https://bugreports.qt.io/browse/QTBUG-40632

---

So which solution should we take here? `CoUninitialize` when `RPC_E_CHANGED_MODE` is returned or just ignore the return status?

---

I'd say ignore the error and don't deal with this crap as long as we test that all COM-related components works fine (Qt, audio, wmf decoder) after all these change.

But do skip CoUnitialize on shutdown if Initialize errored, like what this PR does. This should also apply to the wmf decoder.